### DISCUSSION TITLE
Inheritance and convert

### DISCUSSION THREAD

**Discussion by elodie-28** [Blazebit/blaze-persistence#1926] [Q&A]
Hi everyone,
I'm using Spring Boot with Blaze Persistence.
I have an abstract superclass and two inheriting classes, simplified like this:

```java
public class Entity {
List<Superclass> list;
}

public abstract class Superclass {
}

public class SuperclassChild1 extends Superclass {
}

public class SuperclassChild2 extends Superclass {
}
```

When I use `entityViewManager.convert()` with the Entity to convert to the view I want, the views in the list are converted to the view of the abstract superclass instead of the view corresponding to their "real" class. Is there a way to fix that? I would like to use `entityViewManager.convert()` instead of having to fetch it again from the database.

**Comment by beikov** [comment]
Such a subtype based conversion is not supported yet, but I can see how it's useful. Please create a GitHub issue for this feature request.
